### ISSUE TITLE
autoencoder for LDM

### ISSUE THREAD

**Issue by seung-kim** [CompVis/latent-diffusion#7]
Hi! 
Could you put which autoencoding models correspond to which LDMs on the table, please? 
Maybe I am missing this information somewhere, but it seems it's not clear which one is for which.

**Comment by vvvm23** [comment]
@seung-kim I was struggling with this too. I ran the script `scripts/download_first_stages.sh` which downloaded all the autoencoders, With each autoencoder there is a `config.yaml` file that says the training data was `ldm.data.openimages.FullOpenImagesTrain`. So seems they were all trained on the OpenImages dataset?

@ablattmann @rromb could you please confirm this and also add the information to the README?

**Comment by Eudea** [comment]
> @seung-kim I was struggling with this too. I ran the script `scripts/download_first_stages.sh` which downloaded all the autoencoders, With each autoencoder there is a `config.yaml` file that says the training data was `ldm.data.openimages.FullOpenImagesTrain`. So seems they were all trained on the OpenImages dataset?
> 
> @ablattmann @rromb could you please confirm this and also add the information to the README?

Having the same question, have you fixed it yet?

**Comment by keyu-tian** [comment]
the class `FullOpenImagesTrain` does not exist, maybe the file of ldm/data/openimages.py is missing, could you check that @rromb @ablattmann

**Comment by mia01** [comment]
Hi did anyone figure this out??

**Comment by keyu-tian** [comment]
@mia01 @Eudea @vvvm23 @seung-kim I think im training VQVAEs well on OpenImages. Just with a random crop augmentation (resize to 384 then random crop to 256) and normalizing pixels from [0, 1] to [-1, 1]. For finetuning i use `lr=4e-4, batch_size=1024`. For from scratch i use `lr=4e-6, batch_size=1024`. I use Adam optimizer of `betas=(0.5, 0.9)` following https://github.com/CompVis/taming-transformers/blob/3ba01b241669f5ade541ce990f7650a3b8f65318/taming/models/vqgan.py#L128.

**Comment by wtliao** [comment]
> @mia01 @Eudea @vvvm23 @seung-kim I think im training VQVAEs well on OpenImages. Just with a random crop augmentation (resize to 384 then random crop to 256) and normalizing pixels from [0, 1] to [-1, 1]. For finetuning i use `lr=4e-4, batch_size=1024`. For from scratch i use `lr=4e-6, batch_size=1024`. I use Adam optimizer of `betas=(0.5, 0.9)` following https://github.com/CompVis/taming-transformers/blob/3ba01b241669f5ade541ce990f7650a3b8f65318/taming/models/vqgan.py#L128.

Hi @keyu-tian , I am curious about the distribution of the length of image short side in OpenImages. The vae is trained using augmentation (resize to 384 then random crop to 256), which means all images are downsampled to 384?

**Comment by bu135** [comment]
> @mia01 @Eudea @vvvm23 @seung-kim I think im training VQVAEs well on OpenImages. Just with a random crop augmentation (resize to 384 then random crop to 256) and normalizing pixels from [0, 1] to [-1, 1]. For finetuning i use `lr=4e-4, batch_size=1024`. For from scratch i use `lr=4e-6, batch_size=1024`. I use Adam optimizer of `betas=(0.5, 0.9)` following https://github.com/CompVis/taming-transformers/blob/3ba01b241669f5ade541ce990f7650a3b8f65318/taming/models/vqgan.py#L128.

Hi @keyu-tian . I'm curious if you've done any experiments with VAE instead of VQGAN? I get the impression that the grid effect is hard to eliminate, should the discriminative loss weight be increased?

**Comment by fengyang0317** [comment]
> @mia01 @Eudea @vvvm23 @seung-kim I think im training VQVAEs well on OpenImages. Just with a random crop augmentation (resize to 384 then random crop to 256) and normalizing pixels from [0, 1] to [-1, 1]. For finetuning i use `lr=4e-4, batch_size=1024`. For from scratch i use `lr=4e-6, batch_size=1024`. I use Adam optimizer of `betas=(0.5, 0.9)` following https://github.com/CompVis/taming-transformers/blob/3ba01b241669f5ade541ce990f7650a3b8f65318/taming/models/vqgan.py#L128.

Hi @keyu-tian, can you share the rfid you get and how many step did you run?

**Comment by bhosalems** [comment]
@keyu-tian any idea of how long VAE needs to be trained if I am using on my own dataset? How long did you train or how did you figure this out?

**Comment by zhzer** [comment]
> 如果我在自己的数据集上使用 VAE，知道需要训练多长时间吗？你训练了多长时间，或者你是怎么弄清楚的？

same question
